### ISSUE TITLE
Ability to Prevent Players From Joining Other Waiting Lists When in Waiting List for this One

### ISSUE THREAD

**Issue by TheMattVid** [SkyWars/SkyWars#247]
#### Feature Request:

What would you like to be added?:
I have multiple minigames on one server. Unfortunately, players can join the waiting lists for multiple minigames at once. I'm sure you can figure out the problems that this might cause. Are there any possible solutions that I can follow to prevent this from happening?

**Comment by daboross** [comment]
What other minigames do you use?

I can (probably) add hooks into SkyWars which would remove a player from the SkyWars queue when they enter the queue for another plugin - this would require specific code to hook into each of those other minigame plugins though.

**Comment by TheMattVid** [comment]
As of right now, the only other one is Super Build Battle. Luckily, that plugin already has a feature that disables entering any other commands besides "/bb leave" once you enter the waiting list. Unfortunately, I might add more minigames to the same server, as well, and I'd rather not have to bug you with each one in the future. Is it easily possible to take the feature that disables players from using commands in the skywars game and apply to the waiting list?

**Comment by daboross** [comment]
I think something like that could be added, yeah. It would work best if queuing in the arena (or a waiting area) was also added, but I guess that is another feature.

Right now SkyWars kind of treats the queue as "I'll join a game if one starts", favoring servers which have it as a side-activity rather than a main event. I do want to add the option to have the queue be a full "place" the player is in though, and with that command limiting would also make sense.

**Comment by TheMattVid** [comment]
I agree with that. I have it as a side event too, but since I have multiple side events, it has led to the problem I described.

As a temporary fix, would it be possible for the plugin to force players to issue a command immediately prior to game start? That way, they forfeit the game they are in if that game starts first, but leave the waiting list for that game if Skywars starts first.

**Comment by daboross** [comment]
That sounds reasonable! This would also be more flexible around people joining SkyWars via signs or portals, where another plugin doesn't have the opportunity to block the event.
